subprojects/publish/src/main/java/org/gradle/api/publish/PublishRetrierException.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.publish;
+
+import org.gradle.api.Incubating;
+
+/**
+ *
+ * Exception used for errors associated with {@link PublishRetrier}
+ * @since 6.3
+ */
+@Incubating
+public class PublishRetrierException extends Exception {
```

Comment:
:x:Same comment here, I don't think we should make this public API.

Reply:
Addressed in https://github.com/gradle/gradle/pull/12024/commits/700fac49f84913d6f73d6b87226c6645da89c029